File: Domain/Scenario.cs
namespace Domain
{
  public class Scenario
  {
    public Guid Id { get; set; }
    public string Title { get; set; }
    public DateTime DueDate { get; set; }
    public string Description { get; set; }
    public string Category { get; set; }
    public string BPCycle { get; set; }
    public string File { get; set; }
    public bool IsCancelled { get; set; }
    public ICollection<ScenarioAttendee> Attendees { get; set; }

  }
}

File: client-app/src/app/models/scenario.ts
import { Profile } from "./profile";

export interface Scenario {
  id: string;
  title: string;
  description: string;
  category: string;
  dueDate: Date | null;
  bpCycle: string;
  file: string;
  hostUsername?: string;
  isCancelled?: boolean;
  isGoing?: boolean;
  isHost?: boolean
  attendees?: Profile[]
  host?: Profile;
}

export class ScenarioFormValues {
  id?: string = undefined;
  title: string = '';
  category: string = '';
  description: string = '';
  dueDate: Date | null = null;
  bpCycle: string = '';
  file: string = '';

  constructor(scenario?: ScenarioFormValues) {
    if (scenario) {
      this.id = scenario.id;
      this.title = scenario.title;
      this.category = scenario.category;
      this.description = scenario.description;
      this.dueDate = scenario.dueDate;
      this.bpCycle = scenario.bpCycle;
      this.file = scenario.file;
    }
  }

}

export class Scenario implements Scenario {
  constructor(init?: ScenarioFormValues) {
    Object.assign(this, init);
  }
}